skyportal/handlers/api/candidate/scan_report.py

```diff
@@ -0,0 +1,223 @@
+            )
+            .join(Candidate, Candidate.obj_id == Source.obj_id)
+            .join(Filter)
+            .where(
+                Source.group_id.in_(group_ids),
+                Filter.group_id.in_(group_ids),
+                Source.saved_at.between(
+                    saved_range.get("start_save_date"),
+                    saved_range.get("end_save_date"),
+                ),
+                Candidate.passed_at.between(
+                    detection_range.get("start_date"),
+                    detection_range.get("end_date"),
+                ),
+                Source.active.is_(True),
+            )
+            .group_by(Source.obj_id)
+        ).all()
+    except Exception as e:
+        log(f"Error while retrieving saved candidates: {e}")
+        return []
+
+
+class ScanReportHandler(BaseHandler):
+    @auth_or_token
+    def post(self):
+        """
+        ---
+        summary: Populate the candidate scan report with all saved candidates in a given range
+        tags:
+          - report
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  group_ids:
+                    type: array
+                    items:
+                      type: integer
+                    description: groups use to filter the candidates and manage the report
+                  candidates_detection_range:
+                    type: object
+                    properties:
+                      start_date:
+                        type: string
+                        format: date-time
+                        description: Start date of the candidate detection range
+                      end_date:
+                        type: string
+                        format: date-time
+                        description: End date of the candidate detection range
+                    saved_candidates_range:
+                      type: object
+                      properties:
+                        start_save_date:
+                          type: string
+                          format: date-time
+                          description: Start date of the saved candidates range
+                        end_save_date:
+                          type: string
+                          format: date-time
+                          description: End date of the saved candidates range
+        responses:
+            200:
+                content:
+                  application/json:
+                    schema: Success
+            400:
+                content:
+                  application/json:
+                    schema: Error
+        """
+        data = self.get_json()
+
+        with self.Session() as session:
+            group_ids = data.get("group_ids")
+            if not group_ids:
+                return self.error("No groups provided")
+
+            detection_range = data.get("candidates_detection_range")
```

Comment:
careful here, we are not filtering on the detection time of the candidates! We are filtering on the time at which they passed a user-defined filter